addons/website/static/src/scss/website.scss

```diff
@@ -387,13 +387,35 @@ body {
 
 // Apply background color to the offcanvas menu
 .o_navbar_mobile {
+    $-navbar-mobile-bg-color: o-color('menu-custom') or o-color('menu');
     height: 100dvh;
-    background-color: o-color('menu-custom') or o-color('menu');
+    background-color: $-navbar-mobile-bg-color;
 
     // Make offcanvas close btn above mega menu
     .btn-close {
         z-index: $o-mega-menu-btn-close-zindex;
     }
+
+    .top_menu > .accordion {
+        ul > li > a.nav-link {
+            background-color: inherit;
+        }
+
+        .accordion-button {
+            &:not(.collapsed) {
+                background-color: shade-color($-navbar-mobile-bg-color, 5%);
+            }
+
+            // Replace the default chevron with our Odoo UI icon so it adapts to
+            // the theme colors.
+            &:after {
+                height: 100%;
+                background: none;
+                font-family: 'odoo_ui_icons';
+                content: '\e839';
+            }
+        }
+    }
```

Comment:
Not a fan of those complex very specific selectors either... but ok, maybe there is no other way